internal/ast/js_diagnostics.go

```diff
@@ -0,0 +1,364 @@
+package ast
+
+import (
+	"github.com/microsoft/typescript-go/internal/core"
+	"github.com/microsoft/typescript-go/internal/diagnostics"
+)
+
+// JS syntactic diagnostics functions
+
+func getJSSyntacticDiagnosticsForFile(sourceFile *SourceFile) []*Diagnostic {
+	var diagnostics []*Diagnostic
```

Comment:
Make sure `diagnostics` is always non-nil by doing `diagnostics := []*Diagnostic{}`. This is needed so that the `nil` check in `AdditionalSyntacticDiagnostics` works`.